src/lib/falcor/FalcorRouter.js

```diff
@@ -11,25 +28,169 @@ export default class FalcorRouter extends BaseRouter.createClass([
+        });
+        query += "&fields=slug"
+        pathSet[2].forEach((field, index) => {
+          if (field !== 'slug') {
+            query += "," + ghostify(field);
+          }
+        })
+        query += "&limit=" + pathSet.slugs.length.toString();
+        ghostArticleQuery(query).then((data) => {
+          data = data.posts;
+          const results = [];
+          data.forEach((article, index) => {
+            pathSet[2].forEach((field) => {
+              let ghostField = ghostify(field);
+              results.push({
+                path: ["articlesBySlug", article.slug, field],
+                value: article[ghostField]
+              });
+            });
+          });
+          resolve(results);
+        });
+      });
+    }
+  },
+  {
+    //Get custom article data from SQL database
+    route: "articlesBySlug[{keys:slugs}]['issue', 'category', 'description']",
+    get: (pathSet) => {
+      return new Promise((resolve, reject) => {
+        let query = {slug: {$in: pathSet.slugs}};
+        let projection = {_id: 0};
+        pathSet[2].forEach((field) => {
+          projection[field] = 1;
+        });
+        projection['slug'] = 1;
+        SQLArticleQuery(query, projection).then((data) => {
+          const results = [];
+          data.forEach((article) => {
+            pathSet[2].forEach((field) => {
+              results.push({
+                path: ["articlesBySlug", article.slug, field],
+                value: article[field]
+              });
+            });
+          });
+          resolve(results);
         });
+      });
+    }
+  },
+  {
+    // Get author information from article
+    route: "articlesBySlug[{keys:slugs}]['authors'][{integers:indices}]",
+    get: (pathSet) => {
+      console.log("here");
+      return new Promise((resolve, reject) => {
+        console.log("there");
+        let query = {slug: {$in: pathSet.slugs}};
+        let projection = {_id: 0, slug: 1, authors: 1};
+        SQLArticleQuery(query, projection).then((data) => {
+          console.log("made it and data coming here");
+          const results = [];
+          console.log(data);
+          data.forEach((article) => {
+            let authors = article.authors;
+            let indices = pathSet.indices;
+            for (let i = 0; i < indices.length; i++) {
+              let index = indices[i];
+              if (index >= authors.length) {
+                continue;
+              }
+              results.push({
+                path: ['articlesBySlug', article.slug, 'authors', index],
+                value: $ref(['authorsBySlug', authors[index]])
+              });
+            }
+          });
+          console.log("resolving");
+          console.log(results);
+          resolve(results);
+        });
+      });
+    }
+  },
+  {
+    // Get articles in chronological order by a consecutive range of indices where both min and max are multiples
```

Comment:
Actually Ithought about it some more. The only way this actually makes sense is one of the following:

- 1. fix a global page constant throughout our app, and pages are always that size. The way articlesByPage[1][0] always points to the same article (given nothing new is published). However, in that case we could also equivalently as well just make it articlesByTime[{from: 0, to: 50}] and just make it a programmer task to know to not call more than one page at a time.

2. Make a universal articlesByTime[{from x, to: y}] be allowed and callable. You'll have to work out how to do this efficiently though, and there might not be any efficient way in ghost at all, we probably would have to either fetch redundant data or make multiple fetches. Kinda nasty to program but the simplest API, but prone to making inefficient calls because the working details are masked (and hard to understand without reading the code)

- 3. Have a structure of articlesByPage[pageLength][pageNumber][articleOnPage]. This is a bit clunky API, but it is transparent how it works for a developer consuming the API and pretty easy to program.

The reason we can't have dynamic page sizes without specifying the page size in the path is it breaks the falcor model, and falcor does caching. So what if I call articlesByPage[1] meaning page size 10 and someone else calls articlesByPage[1] meaning page size 20. I will get bad results in my second call cause some of the results will be cached, and overall it will make some very horrible bugs.

Overall I'm a fan of approach 3, followed by 1, followed by 2.